internal/providers/pluginfw/products/permissions/resource_permission.go

```diff
@@ -0,0 +1,652 @@
+		)
+		return
+	}
+
+	// Get the mapping from the ID to determine object type and field
+	mapping, err := permissions.GetResourcePermissionsFromId(objectID)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get resource permissions mapping", err.Error())
+		return
+	}
+
+	// Read current permissions from Databricks
+	idParts := strings.Split(objectID, "/")
+	permID := idParts[len(idParts)-1]
+
+	perms, err := w.Permissions.Get(ctx, iam.GetPermissionRequest{
+		RequestObjectId:   permID,
+		RequestObjectType: mapping.GetRequestObjectType(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to read permissions", err.Error())
+		return
+	}
+
+	// Find the specific principal's permission
+	var found bool
+	var permissionLevel string
+	var userName, groupName, servicePrincipalName string
+
+	for _, acl := range perms.AccessControlList {
+		if r.matchesPrincipal(acl, principal) {
+			if len(acl.AllPermissions) > 0 {
+				permissionLevel = string(acl.AllPermissions[0].PermissionLevel)
+				userName = acl.UserName
+				groupName = acl.GroupName
+				servicePrincipalName = acl.ServicePrincipalName
+				found = true
+				break
+			}
+		}
+	}
+
+	if !found {
+		resp.Diagnostics.AddError(
+			"Permission Not Found",
+			fmt.Sprintf("No permission found for principal %q on object %q", principal, objectID),
+		)
+		return
+	}
+
+	// Set all attributes in state using SetAttribute
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), types.StringValue(req.ID))...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("object_type"), types.StringValue(mapping.GetRequestObjectType()))...)
+
+	// Set the object identifier field (e.g., cluster_id, job_id, etc.)
+	// Extract the configured value from the objectID
+	configuredValue := permID
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(mapping.GetField()), types.StringValue(configuredValue))...)
+
+	// Set principal fields - use null for empty strings to avoid ImportStateVerify failures
+	if userName != "" {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("user_name"), types.StringValue(userName))...)
+	} else {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("user_name"), types.StringNull())...)
+	}
+
+	if groupName != "" {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("group_name"), types.StringValue(groupName))...)
+	} else {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("group_name"), types.StringNull())...)
+	}
+
+	if servicePrincipalName != "" {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("service_principal_name"), types.StringValue(servicePrincipalName))...)
+	} else {
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("service_principal_name"), types.StringNull())...)
+	}
+
+	// Set permission level
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("permission_level"), types.StringValue(permissionLevel))...)
```

Comment:
Should we populate an instance of PermissionResourceModel and then write the entire resource to the plan?